documenteer/stackdocs/build.py

```diff
@@ -44,3 +49,128 @@ def discover_setup_packages():
         packages[name] = info
 
     return packages
+
+
+def find_package_docs(package_dir):
+    """Find documentation directories in a package using ``manifest.yaml``.
+
+    Parameters
+    ----------
+    package_dir : `str`
+        Directory of an EUPS package.
+
+    Returns
+    -------
+    doc_dirs : namedtuple
+        Attributes of the named tuple are:
+
+        - ``package_dirs`` (`dict`). Keys are package names (for example,
+          ``'afw'``). Values are absolute directory paths to the package's
+          documentation directory inside the package's ``doc`` directory. If
+          there is no package-level documentation the dictionary will be empty.
+
+        - ``modules_dirs`` (`dict`). Keys are module names (for example,
+          ``'afw.table'``). Values are absolute directory paths to the module's
+          directory inside the package's ``doc`` directory. If a package has no
+          modules the returned dictionary will be empty.
```

Comment:
Yup, we're organizing content in `doc/` into a flat set of directories for Python modules and the package. I'm prototyping this out in `pipe_base` right now: https://github.com/lsst/pipe_base/tree/tickets/DM-11253/doc